backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -28,16 +33,87 @@ public DefaultUserService(UserDao userDao) {
 
     @Override
     public User getUserById(Long id) {
+        LOG.info("Retrieved user ID {}", id);
         return userDao.findUserById(id);
     }
 
+    /**
+     * Find a unique User with email and username
+     * @param email The email of the user
+     * @param username The username of the user
+     * @return The User in the database
+     */
+    @Override
+    public User getUserByEmailAndUsername(String email, String username) {
+        List<User> userList = userDao.findAll();
+        User userToFind = null;
+
+        for (User user : userList) {
+            if (email.equalsIgnoreCase(user.getEmail()) && username.equalsIgnoreCase(user.getUsername())) {
+                userToFind = user;
+                LOG.info("Found user with email {}", user.getEmail());
+                break;
+            }
+        }
+
+        return userToFind;
+    }
+
     @Override
     public User createUser(User user) {
+        String password = user.getPassword();
+        String encryptedPassword = encryptData(password);
+        user.setPassword(encryptedPassword);
+        userDao.save(user);
+        LOG.info("Created user: {}", user.getFirstName());
+
+        return user;
+    }
 
-        User createdUser = userDao.save(user);
+    /**
+     * Method to look if user exists with unique email and username.
+     * @param email The email coming from the frontend.
+     * @param username The username coming from the frontend.
+     * @return boolean if exists or not.
+     */
+    @Override
+    public boolean userExists(String email, String username) {
+        boolean exists = false;
+        List<User> userList = userDao.findAll();
```

Comment:
We could do `userDao.findUserByEmail();` and if it returns something then user exists

Reply:
True. This is even better performance. Thanks!